### DISCUSSION TITLE
Authorize payload in third argument of setter

### DISCUSSION THREAD

**Discussion by stephanepx** [pmndrs/zustand#1194] [Q&A]
Hi everybody! 

I want to see payload in devtools middleware, but it seems that TS does not accept other props than `type` in third argument of the setter function : 

``` typescript
const createYoloSlice: StateCreator<
  ZustandStore,
  [],
  [],
  YoloSlice
> = set => ({
  setSomething: something =>
    set(
      state => {
        state.yolo.something = something;
      },
      false,
      // here I have to use @ts-ignore to pass payload
      { payload: something, type: 'yolo/setSomething' },
    ),
  something: 0,
});
```

Do you have any idea how I can handle this nicely ?

Thanks!!

**Comment by dai-shi** [comment]
Here you go: https://tsplay.dev/WYZJxw

Please wait for #1183 to fix the error.

**Reply by stephanepx** [reply]
Thanks for your reply ! I forgot to precise that I used `immer` for my example. 
Do you know if there is a way to directly have payload and type included in all setter functions in all slices by just passing a prefix like `yolo/` ?

**Reply by dai-shi** [reply]
Do you mean like #1025?
[zustand-lens](https://github.com/dhmk083/dhmk-zustand-lens) seems a promising solution in this field.

**Reply by stephanepx** [reply]
Hum not sure.. I want to have in devTools all my functions with names (no anonymous) and payload
![image](https://user-images.githubusercontent.com/88713643/185084643-d3338c43-745a-4b24-8920-1d4a6ae28995.png)

**Reply by dai-shi** [reply]
Sorry, it was my misunderstanding.
#1185 is a relevant discussion, and the answer is it's a field for 3rd-party libraries.

**Reply by stephanepx** [reply]
Thanks!
